src/Compilers/CSharp/Portable/Symbols/TypeWithAnnotations.cs

```diff
@@ -50,13 +51,28 @@ internal struct Builder
             /// </remarks>
             internal bool InterlockedInitialize(TypeWithAnnotations type)
             {
-                if ((object)_defaultType != null)
+                if (!IsDefault)
                 {
                     return false;
                 }
-                _nullableAnnotation = type.NullableAnnotation;
-                Interlocked.CompareExchange(ref _extensions, type._extensions, null);
-                return (object)Interlocked.CompareExchange(ref _defaultType, type.DefaultType, null) == null;
+                Interlocked.CompareExchange(ref _nullableAnnotation, (int)type.NullableAnnotation, 0);
+                Interlocked.CompareExchange(ref _defaultType, type.DefaultType, null);
+                bool wasFirst = Interlocked.CompareExchange(ref _extensions, type._extensions ?? Extensions.Default, null) is null;
```

Comment:
The current pattern requires that _extensions be the last to be initialized.  I think it'd be worth a comment here highlighting that reordering these in the source would be potentially problematic.